src/util/translations.h

```diff
@@ -13,92 +13,101 @@ namespace mixxx {
 class Translations {
   public:
     static void initializeTranslations(UserSettingsPointer pConfig,
-                                       QCoreApplication* pApp,
-                                       const QString& forcedLocale) {
-        QString resourcePath = pConfig->getResourcePath();
-        QString translationsFolder = resourcePath + "translations/";
+            QCoreApplication* pApp,
+            const QString& cmdlineLocaleString) {
+        const QString translationsPath =
+                pConfig->getResourcePath() + QStringLiteral("translations/");
 
-        // Load Qt base translations
-        QString userLocale = forcedLocale;
-        QLocale systemLocale = QLocale::system();
+        // Set the default locale, if specified via command line or config
+        // file. If neither is the case, the default locale will be the
+        // system's locale.
+        {
+            QString customLocaleString = QString();
+            if (!cmdlineLocaleString.isEmpty()) {
+                customLocaleString = cmdlineLocaleString;
+            } else {
+                const QString configLocale = pConfig->getValueString(
+                        ConfigKey("[Config]", "Locale"));
+                if (!configLocale.isEmpty()) {
+                    customLocaleString = configLocale;
+                }
+            }
 
-        // Attempt to load user locale from config
-        if (userLocale.isEmpty()) {
-            userLocale = pConfig->getValueString(ConfigKey("[Config]", "Locale"));
+            if (!customLocaleString.isEmpty()) {
+                const QLocale customLocale = QLocale(customLocaleString);
+                // If the customLocaleString is not a valid locale, Qt will
+                // automatically use the "C" locale instead. In that case,
+                // let's print a warning.
+                if (customLocaleString.compare(
+                            QStringLiteral("C"), Qt::CaseInsensitive) != 0 &&
+                        customLocale.language() == QLocale::C) {
+                    qWarning() << "Custom locale not found, using 'C' locale "
+                                  "(check your configuration to avoid this "
+                                  "warning).";
+                }
+                QLocale::setDefault(customLocale);
+            }
         }
 
-        if (userLocale.isEmpty()) {
-            QLocale::setDefault(QLocale(systemLocale));
-        } else {
-            QLocale::setDefault(QLocale(userLocale));
+        QLocale locale = QLocale();
```

Comment:
This assignment is redundant.
Please add a comment that we investigate the default value.

Reply:
> Please add a comment that we investigate the default value.

What do you mean?

From the [Qt docs](https://doc.qt.io/qt-5/qlocale.html#QLocale):
> Constructs a QLocale object initialized with the default locale. If no default locale was set using setDefault(), this locale will be the same as the one returned by system().

Reply:
I added that as a comment.